src/components/Referrals/ConfirmInviteUser.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <dialog-base :visible.sync="isVisible" :show-close-button="false" custom-class="dialog--confirm-invite-user">
+    <div class="invite-user-icon">
+      <s-icon name="file-file-text-24" size="40px" />
+    </div>
+    <p class="invite-user-ititle">{{ t('referralProgram.confirm.inviteTitle') }}</p>
+    <p class="invite-user-description">{{ t('referralProgram.confirm.inviteDescription') }}</p>
+    <template #footer>
+      <s-button type="primary" class="s-typography-button--large" :disabled="loading" @click="handleConfirmInviteUser">
+        {{ t('referralProgram.confirm.signInvitation') }}
+      </s-button>
+      <div class="invite-user-free-charge">
+        <s-icon class="invite-user-info" name="chevron-down-rounded-16" size="14px" />
+        <span>{{ t('referralProgram.confirm.freeOfCharge') }}</span>
+      </div>
+    </template>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator';
+import { Getter, Action } from 'vuex-class';
+import { api, mixins } from '@soramitsu/soraneo-wallet-web';
+
+import DialogMixin from '@/components/mixins/DialogMixin';
+import DialogBase from '@/components/DialogBase.vue';
+
+@Component({
+  components: { DialogBase },
+})
+export default class ConfirmInviteUser extends Mixins(mixins.TransactionMixin, DialogMixin) {
+  @Getter storageReferral!: string;
+
+  @Action setReferral!: (value: string) => Promise<void>;
+
+  async handleConfirmInviteUser(): Promise<void> {
+    try {
+      await this.withNotifications(async () => await api.referralSystem.setInvitedUser(this.storageReferral));
+      this.$emit('confirm', true);
+    } catch (error) {
+      this.$emit('confirm');
+    }
+    this.setReferral('');
```

Comment:
For me as user it's look a little strange, that i should click approve button and then not sign transaction, to reset my referee. If i will just close invite modal, the referee is still be in storage, and every page reload i will see this modal...

Reply:
You are right! I've changed behaviour.